client/landing/gutenboarding/stores/onboard/reducer.ts

```diff
@@ -1,113 +1,87 @@
 /**
  * External dependencies
  */
-import { Reducer } from 'redux';
 import { combineReducers } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
-import { ActionType, SiteVertical } from './types';
-import * as Actions from './actions';
+import { SiteVertical } from './types';
+import { OnboardAction } from './actions';
 
-const domain: Reducer<
-	import('@automattic/data-stores').DomainSuggestions.DomainSuggestion | undefined,
-	ReturnType< typeof Actions[ 'setDomain' ] | typeof Actions[ 'resetOnboardStore' ] >
-> = ( state = undefined, action ) => {
-	if ( action.type === ActionType.SET_DOMAIN ) {
+type DomainSuggestion = import('@automattic/data-stores').DomainSuggestions.DomainSuggestion;
+type Template = import('@automattic/data-stores').VerticalsTemplates.Template;
+
+function domain( state: DomainSuggestion | undefined, action: OnboardAction ) {
```

Comment:
I like using the `Reducer` type because it ensures we return a state type. I'd say it's a bit more cumbersome here using `function` declarations:

```suggestion
function domain( state: DomainSuggestion | undefined, action: OnboardAction ): DomainSuggestion {
```

If we stay with `const` declarations and `Reducer`, we get the following which seems like a good place:

```suggestion
const domain: Reducer<DomainSuggestion | undefined, OnboardAction> = ( state, action ) => {
```

If we don't return the same type of state, that's a bug types can help detect.

Reply:
I've dropped the commit that converted the reducers to function expressons. Given all selectors and action creators are using the `const` way it doesn't make sense for the reducers to be any different.

> I'd say it's a bit more cumbersome here using function declarations

Given that I think it's less cumbersome this way goes to show it must be a preference thing, so better to just go with the flow of what's already there.

Then again I _did_ forget to specify the return types, so maybe you have a point 😅

Reply:
I don't have much of a preference as long as we're getting well typed reducers (with return):

```ts
const r: Reducer< State, Action > = ( s = initialValue, a ) => s;
// or
function r_( s: State = initialValue, a: Action ): State { return s; }